extensions/default/src/DicomWebDataSource/utils/clientManager.ts

```diff
@@ -0,0 +1,215 @@
+import { api } from 'dicomweb-client';
+import StaticWadoClient from './StaticWadoClient';
+import dcm4cheeReject from '../dcm4cheeReject';
+import { errorHandler, utils } from '@ohif/core';
+
+export default class ClientManager {
+  clients;
+  userAuthenticationService;
+
+  constructor({ params, query, dicomWebConfig, userAuthenticationService }) {
+    this.clients = [];
+    this.userAuthenticationService = userAuthenticationService;
+    if (Array.isArray(dicomWebConfig)) {
+      dicomWebConfig.forEach(config => this.addConfiguration(params, query, config));
+    } else {
+      this.addConfiguration(params, query, dicomWebConfig);
+    }
+  }
+
+  /**
+   * Adds a dicomweb server configuration in the clients list
+   * @param configToAdd
+   * @returns {void}
+   */
+  private _addConfiguration(configToAdd): void {
+    const config = Object.assign({}, configToAdd);
+    config.qidoConfig = {
+      url: config.qidoRoot,
+      staticWado: config.staticWado,
+      singlepart: config.singlepart,
+      headers: this.userAuthenticationService.getAuthorizationHeader(),
```

Comment:
this is getting called more than necessary

this.userAuthenticationService.getAuthorizationHeader()
errorHandler.getErroHandler() too I think

Reply:
Does each call creates a different object? Isn't safe this way?